ios-base/Networking/Services/APIClient.swift

```diff
@@ -0,0 +1,341 @@
+    HTTPHeader.accept.rawValue: "application/json",
+    HTTPHeader.contentType.rawValue: "application/json"
+  ]
+  
+  fileprivate class func getHeaders() -> [String: String] {
+    if let session = SessionManager.currentSession {
+      return baseHeaders + [
+        HTTPHeader.uid.rawValue: session.uid ?? "",
+        HTTPHeader.client.rawValue: session.client ?? "",
+        HTTPHeader.token.rawValue: session.accessToken ?? ""
+      ]
+    }
+    return baseHeaders
+  }
+  
+  fileprivate class func getBaseUrl() -> String {
+    return Bundle.main.object(forInfoDictionaryKey: "Base URL") as? String ?? ""
+  }
+  
+  //Recursively build multipart params to send along with media in upload requests.
+  //If params includes the desired root key,
+  //call this method with an empty String for rootKey param.
+  class func multipartFormData(
+    _ multipartForm: MultipartFormData,
+    params: Any,
+    rootKey: String
+  ) {
+    switch params.self {
+    case let array as [Any]:
+      for val in array {
+        let forwardRootKey = rootKey.isEmpty ? "array[]" : rootKey + "[]"
+        multipartFormData(multipartForm, params: val, rootKey: forwardRootKey)
+      }
+    case let dict as [String: Any]:
+      for (key, value) in dict {
+        let forwardRootKey = rootKey.isEmpty ? key : rootKey + "[\(key)]"
+        multipartFormData(multipartForm, params: value, rootKey: forwardRootKey)
+      }
+    default:
+      if let uploadData = "\(params)".data(
+        using: String.Encoding.utf8,
+        allowLossyConversion: false
+      ) {
+        let forwardRootKey = rootKey.isEmpty ?
+          "\(type(of: params))".lowercased() : rootKey
+        multipartForm.append(uploadData, withName: forwardRootKey)
+      }
+    }
+  }
+  
+  //Multipart-form base request. Used to upload media along with desired params.
+  //Note: Multipart request does not support Content-Type = application/json.
+  //If your API requires this header
+  //do not use this method or change backend to skip this validation.
+  class func multipartRequest(_ method: HTTPMethod = .post,
+                              url: String,
+                              headers: [String: String] = APIClient.getHeaders(),
+                              params: [String: Any]?,
+                              paramsRootKey: String,
+                              media: [MultipartMedia],
+                              success: @escaping SuccessCallback,
+                              failure: @escaping FailureCallback) {
+    
+    let requestConvertible = BaseURLRequestConvertible(
+      path: url,
+      method: method,
+      headers: headers
+    )
+    
+    AF.upload(
+      multipartFormData: { (multipartForm) -> Void in
+        if let parameters = params {
+          multipartFormData(multipartForm, params: parameters, rootKey: paramsRootKey)
+        }
+        for elem in media {
+          elem.embed(inForm: multipartForm)
+        }
+    },
+      with: requestConvertible)
```

Comment:
I will close the method call in a new line so the subsequent call don't appear at the same level that the previous call parameter